boot/assets.go

```diff
@@ -513,8 +527,12 @@ func (o *TrustedAssetsUpdateObserver) Observe(op gadget.ContentOperation, partRo
 	}
 	if o.modeenv == nil {
 		// we've hit a trusted asset, so a modeenv is needed now too
+		modeenvLock()
+		o.modeenvLocked = true
 		o.modeenv, err = ReadModeenv("")
```

Comment:
Should this call `loadModeenv` instead? Even though, it is kind of useless, when looking for all `ReadModeenv` calls not in tests, this one shows up. But this one is safe. That makes reviewing more obvious.